src/java.base/share/classes/java/io/BufferedReader.java

```diff
@@ -394,12 +398,74 @@ String readLine(boolean ignoreLF, boolean[] term) throws IOException {
      *
      * @throws     IOException  If an I/O error occurs
      *
+     * @see #readAllLines
      * @see java.nio.file.Files#readAllLines
      */
     public String readLine() throws IOException {
         return readLine(false, null);
     }
 
+    /**
+     * Reads all remaining lines of text. A line is considered to be terminated
+     * by any one of a line feed ('\n'), a carriage return ('\r'), a carriage
+     * return followed immediately by a line feed, or by reaching the
+     * end-of-file (EOF).
+     *
+     * <p> This method works as if invoking it were equivalent to evaluating
+     * the expression:
+     * <blockquote>{@link #lines()}.toList()</blockquote>
+     *
+     * @apiNote
+     * This method is intended for simple cases where it is convenient
+     * to read all lines in a single operation. It is not intended for
+     * reading a large number of lines.
+     *
+     * @return     the lines of text as a {@code List}; whether the
+     *             {@code List} is modifiable is implementation dependent,
+     *             hence not specified
+     *
```

Comment:
Does the rationale used in Files to leaving this indeterminate apply here?
I'd think it would be better to say its not-modifyable.
Here its free/cheap, the Stream<T>.toList() method creates an un-modifiableList.

Reply:
Please see the changes in [4fac584](https://github.com/openjdk/jdk/pull/24728/commits/4fac584c5b19b3910e7a362656d2b406ad4ba189).